swim/labels.go

```diff
@@ -1,5 +1,154 @@
 package swim
 
+import (
+	"errors"
+	"strings"
+
+	"github.com/uber/ringpop-go/util"
+)
+
+var (
+	// DefaultLabelOptions contain the default values to be used to limit the
+	// amount of data being gossipped over the network. The defaults have been
+	// chosen with the following assumptions.
+	// 1. 1000 node cluster
+	// 2. Every byte available is used (yes, characters are not bytes but these
+	//    are ballpark figures to protect developers)
+	// 3. Worst case would have continues fullsync's, meaning the complete
+	//    memberlist being sent over the wire 5 times a second.
+	// When all contiditions are met the Labels would add the following load
+	// (non-compressed) to the network:
+	//    (32+128)*5*1000*5*8 = ~32mbit
+	DefaultLabelOptions = LabelOptions{
+		KeySize:   32,
+		ValueSize: 128,
+		Count:     5,
+	}
+
+	// ErrLabelSizeExceeded indicates that an operation on labels would exceed
+	// the configured size limits on labels. This is to prevent applications
+	// bloating the gossip protocol with too much data. Ringpop can be
+	// configured with the settings to control the amount of data that can be
+	// put in a nodes labels.
+	ErrLabelSizeExceeded = errors.New("label operation exceeds configured label limits")
+
+	labelsPrivateNamespacePrefix = "__"
+)
+
+// LabelOptions controlls the limits on labels. Since labels are gossiped on
+// every ping/ping-req/fullsync we need to limit the amount of data an
+// application stores in their labels. When needed the defaults can be
+// overwritten during the construction of ringpop. This should be done with care
+// to not overwhelm the network with data.
+type LabelOptions struct {
```

Comment:
By changing these value's to `uint` it is non-trivial to compare the numbers to the return value of `len()` as you can see with this test in `gore`.

```
$ gore
gore version 0.2.5  :help for help
gore> var ui uint
gore> var m map[string]string
gore> ui > len(m)
# command-line-arguments
/var/folders/9q/ntq49_lj1_39ry71kwl5x8480000gn/T/347859520/gore_session.go:10: invalid operation: ui > len(m) (mismatched types uint and int)
error: exit status 2
exit status 2
gore>
```